components/sensor/collector.go

```diff
@@ -0,0 +1,69 @@
+package sensor
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"go.viam.com/rdk/data"
+)
+
+type method int64
+
+const (
+	readings method = iota
+)
+
+func (m method) String() string {
+	if m == readings {
+		return "Readings"
+	}
+	return "Unknown"
+}
+
+// ReadingRecords a collection of ReadingRecord.
+type ReadingRecords struct {
+	Readings []ReadingRecord
+}
+
+// ReadingRecord a single analog reading.
+type ReadingRecord struct {
+	ReadingName string
+	Reading     interface{}
+}
+
+func newSensorCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	sensorResource, err := assertSensor(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	cFunc := data.CaptureFunc(func(ctx context.Context, arg map[string]*anypb.Any) (interface{}, error) {
+		var records []ReadingRecord
+		values, err := sensorResource.Readings(ctx, nil) // TODO: pass in something here from the config rather than nil?
```

Comment:
If there are any readers that require `extra` parameters, should support grabbing (and validating that is a map type?) `args['extra']` if non-nil. Can make this into a ticket to be nonblocking for this PR.

Reply:
I was thinking about that, the reason I decided not to for this PR was that this would interact with a sensor that returned a reading whose name was "extra"; I didn't want to be the one making that call since this PR reaches outside my usual scope of code changes.

But if that's the way that you think this should work I'm happy to include in this PR.

Reply:
I want to redesign how we're currently passing in reader names (e.g. support nested `additional_params`), and will include this in that refactor. If you could create a ticket for this and link to it in the TODO, that'd be helpful. Thank you!

Reply:
Done